### DISCUSSION TITLE
UX...

### DISCUSSION THREAD

**Discussion by metaleap** [AchetaGames/Epic-Asset-Manager#286] [Q&A] [answered]
So what I actually wanna do is to bring all my (over 200) assets onto my (big enough) disk in one go, so I can scrutinize them quickly whenever needed (without needing to wait for download later on, when actually being in flow of picking/eval'ing from my available assets for the project at hand).

Well when I download them with your (great btw) tool, each lands in a folder named sth like eg. M1973152c58dbae466baV2. I cannot select+copy the _actual title_ of that asset in this tool (ie to easily quickly manually rename such an "ID-named" downloaded-asset folder after download). 

So I thought maybe I'll code up a prog to rename every such asset-download folder afterwards, using the metadata that might be found in each download's `manifest.json` file. Well guess what, even just the friggin Asset Title is just not in there! It's just packaging/download info stuff in there, not asset info/metadata... (That's on Epic, of course, not on @aknarts  =)

Any ideas how I can bulk-download all my stuff while keeping it all properly named? I was prepared to spend a day even doing this basically manually, just to get it all "stashed and ready anytime" locally — if only I could at least select and copy each asset's title. But I can't..  :face_with_spiral_eyes: 

Does this Gnome GUI stuff you're using allow text selection and ctrl+C / copy-to-clipboard capabilities? Would be a real boon for use-cases beyond just-a-one-off-download!

**Comment by aknarts** [comment, marked as answer]
So the download manifest is just that, it is a manifest of the data to be downloaded and the directory name is based on the versioning name that comes directly from upstream(also it matches how Epic downloads it on Windows so the two vault should be compatible).

As for the name of the asset that can be gotten from the asset info, that lives in your cache for flatpak it would be in `~/.var/app/io.github.achetagames.epic_asset_manager/cache/epic_asset_manager/`

And I think the GTK allows the select&copy workflow but it needs to be explicitly enabled for the element. I can look into enabling that. 

By the way if you are ok with doing some rust the [underlying library](https://github.com/AchetaGames/egs-api-rs) might be a good point to start the mass download of assets.

**Comment by metaleap** [comment]
> And I think the GTK allows the select&copy workflow but it needs to be explicitly enabled for the element. I can look into enabling that.

No need for text-selection actually, if only there was a per-asset action "Copy Title", eg. next to "Download" and "Add To Project"  =)

> As for the name of the asset that can be gotten from the asset info, that lives in your cache for flatpak it would be in ~/.var/app/io.github.achetagames.epic_asset_manager/cache/epic_asset_manager/

Great pointer, will take a lookie, thanks!
